### PR TITLE
internal/namingschema: add config flag to disable default integration service names

### PR DESCRIPTION
**Pull request by rarguelloF**
### What does this PR do?

- Adds `DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED` env variable and `WithGlobalServiceName` tracer start option. 

### Motivation

The motivation of this flag is to be able to to use the default service names introduced by the naming schema v1 (everything defaults to `DD_SERVICE`) while using naming schema v0 (so the operation names remain unchanged). 

AKA: This allows customers to upgrade to v1 SERVICE names while maintaining their V0 operation names. (Otherwise, V1 changes both service and operation names)

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality.
- [x] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Review by zarirhamza — APPROVED** [review]
LGTM

**Review by amarziali — DISMISSED** [review]
lgtm

File: internal/namingschema/namingschema.go [hunk 1]
```diff
@@ -69,6 +72,20 @@ func SetDefaultVersion() Version {
...
+func GetRemoveIntegrationServiceNamesEnabled() bool {
+	rmIntegrationServiceNamesMu.RLock()
+	defer rmIntegrationServiceNamesMu.RUnlock()
+	return rmIntegrationServiceNames
+}
+
+// SetRemoveIntegrationServiceNamesEnabled sets the value of the RemoveIntegrationServiceNames setting used for this application.
+func SetRemoveIntegrationServiceNamesEnabled(v bool) {
+	rmIntegrationServiceNamesMu.RLock()
+	defer rmIntegrationServiceNamesMu.RUnlock()
```
Comment (pkalmakis) [inline internal/namingschema/namingschema.go:85]:
```suggestion
	rmIntegrationServiceNamesMu.Lock()
	defer rmIntegrationServiceNamesMu.Unlock()
```

Comment (rarguelloF) [reply internal/namingschema/namingschema.go:85]: thanks for catching this up!

File: ddtrace/tracer/option.go [hunk 2]
```diff
@@ -234,6 +234,9 @@ func newConfig(opts ...StartOption) *config {
 		c.spanAttributeSchemaVersion = int(v)
 		log.Warn("DD_TRACE_SPAN_ATTRIBUTE_SCHEMA=%s is not a valid value, setting to default of v%d", schemaVersionStr, v)
 	}
+	// Allow DD_TRACE_SPAN_ATTRIBUTE_SCHEMA=v0 users to disable default integration (contrib) service names.
```
Comment (ajgajg1134) [inline ddtrace/tracer/option.go:237]:
```suggestion
	// Allow DD_TRACE_SPAN_ATTRIBUTE_SCHEMA=v0 users to disable default integration (contrib AKA v0) service names.
```

File: internal/namingschema/service_name_test.go [hunk 3]
```diff
@@ -60,19 +61,50 @@ func TestNewDefaultServiceName(t *testing.T) {
...
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			version := namingschema.GetVersion()
 			defer namingschema.SetVersion(version)
 			namingschema.SetVersion(tc.schemaVersion)
 
+			if tc.beforeTestHook != nil {
+				cleanup := tc.beforeTestHook(t)
+				defer cleanup()
```
Comment (knusbaum) [inline internal/namingschema/service_name_test.go:101]:
```suggestion
				defer tc.beforeTestHook(t)()
```

File: internal/namingschema/service_name_test.go [hunk 4]
```diff
@@ -19,11 +19,12 @@ func TestNewDefaultServiceName(t *testing.T) {
...
 	testCases := []struct {
-		name          string
-		schemaVersion namingschema.Version
-		ddService     string
-		opts          []namingschema.Option
-		want          string
+		name           string
+		schemaVersion  namingschema.Version
+		ddService      string
+		beforeTestHook func(t *testing.T) func()
```
Comment (knusbaum) [inline internal/namingschema/service_name_test.go:25]:
```suggestion
		beforeTest func() func()
```

File: internal/namingschema/service_name_test.go [hunk 5]
```diff
@@ -60,19 +61,50 @@ func TestNewDefaultServiceName(t *testing.T) {
 			opts:          nil,
 			want:          "dd-service",
 		},
+		{
+			name:          "schema v0 without rm integration service names",
+			schemaVersion: namingschema.SchemaV0,
+			ddService:     "dd-service",
+			beforeTestHook: func(_ *testing.T) func() {
```
Comment (knusbaum) [inline internal/namingschema/service_name_test.go:68]: `testing.T` is not used anywhere in the `beforeTestHook`s. Let's get rid of it.

**Review by ajgajg1134 — DISMISSED** [review]

**Review by knusbaum — DISMISSED** [review]

File: internal/namingschema/namingschema.go [hunk 6]
```diff
@@ -72,18 +72,18 @@ func SetDefaultVersion() Version {
...
-// SetRemoveIntegrationServiceNamesEnabled sets the value of the RemoveIntegrationServiceNames setting used for this application.
-func SetRemoveIntegrationServiceNamesEnabled(v bool) {
-	rmIntegrationServiceNamesMu.Lock()
-	defer rmIntegrationServiceNamesMu.Unlock()
-	rmIntegrationServiceNames = v
+// SetUseGlobalServiceName sets the value of the RemoveIntegrationServiceNames setting used for this application.
+func SetUseGlobalServiceName(v bool) {
+	useGlobalServiceNameMu.Lock()
+	defer useGlobalServiceNameMu.Unlock()
+	useGlobalServiceName = v
```
Comment (dianashevchenko) [inline internal/namingschema/namingschema.go]: I believe it's not supposed to be RemoveIntegrationServiceNames in the comment description

**Review by dianashevchenko — APPROVED** [review]

**Pull request merged**